client/src/stores/history/historyItemsStore.js

```diff
@@ -21,22 +21,24 @@ export const useHistoryItemsStore = defineStore("historyItemsStore", {
         latestCreateTime: new Date(),
         totalMatchesCount: undefined,
         lastCheckedTime: new Date(),
-        matchedHids: {},
+        relatedItems: {},
         isWatching: false,
     }),
     getters: {
         getHistoryItems: (state) => {
             return (historyId, filterText) => {
                 const itemArray = state.items[historyId] || [];
                 const filters = HistoryFilters.getFilters(filterText).filter((filter) => !filter.includes("related"));
+                const relatedHid = HistoryFilters.getQueryDict(filterText)["related-eq"] || null;
```

Comment:
We should add another filter helper to just grab a particular filter value, otherwise `getQueryDict` will unpack and convert all filter values first.

Reply:
Done! Added a `getFilterValue` function.